cloud_foundry_api/datadog_checks/cloud_foundry_api/cloud_foundry_api.py

```diff
@@ -198,26 +330,113 @@ def get_events(self, additional_tags):
                 "order-by": "timestamp",
             }
             headers = {"Authorization": "Bearer {}".format(self._oauth_token)}
-            url = "{}/v2/events".format(self._api_url)
-            return self.scroll_pages(url, params, headers, additional_tags)
+            url = join_url(self._api_url, "v2/events")
+            return self.scroll_events(url, params, headers)
         elif self._api_version == "v3":
             params = {
                 "types": self._event_filter,
                 "per_page": min(self._per_page, MAX_PAGE_SIZE_V3),
                 "order_by": "-created_at",
             }
             headers = {"Authorization": "Bearer {}".format(self._oauth_token)}
-            # Make sure to have at least one trailing slash to avoid surprises with urljoin
-            url = urljoin(self._api_url + "/", "v3/audit_events")
-            return self.scroll_pages(url, params, headers, additional_tags)
+            url = join_url(self._api_url, "v3/audit_events")
+            return self.scroll_events(url, params, headers)
 
         self.log.error("Unknown api version `%s`", self._api_version)
         return {}
 
+    def parse_event(self, cf_event):
+        # type: (Dict[str, Any]) -> Tuple[Event, str, int]
+        dd_event = {}
+        event_guid = ""
+        event_ts = 0
+
+        if self._api_version == "v2":
```

Comment:
It's a shame that you have to put back that code in the check itself. I like that you started hiding some of the v2/v3 differences. Ideally, all of it would be hidden in the check class, and we'd use some utilities. This is probably a bigger refactoring that can be done later though.